packages/rspack/src/Module.ts

```diff
@@ -48,26 +49,33 @@ export class Module {
 	#inner: JsModule;
 	_originalSource?: Source;
 
+	context?: string;
+	resource?: string;
+	request?: string;
+	userRequest?: string;
 	rawRequest?: string;
 
-	static __from_binding(module: JsModule) {
-		return new Module(module);
+	buildInfo: Record<string, any>;
```

Comment:
Please add some explanation to the mutability of the field. It looks like this does not affect the rust side right?

Reply:
Yes. I have observed that when plugins or loaders dynamically add fields to `buildInfo` or `buildMeta`, these fields are primarily used as markers and are utilized later on the JavaScript side. Therefore, there is no need to pass them back to Rust. 
For the fields that `buildInfo` inherently has and that are related to the build, perhaps they can be exposed and written back in the future version. And by using a Proxy to combine the two, this could be transparent to plugin developers.